astropy/io/fits/connect.py

```diff
@@ -302,3 +306,202 @@ def write_table_fits(input, output, overwrite=False):
+        ``uncertainty``.
+        Default is ``0`` (data), ``0`` (meta), ``'mask'`` (mask) and
+        ``'uncert'`` (uncertainty).
+
+    kw_unit : str or None, optional
+        The header keyword which translates to the unit for the data. Set it
+        to ``None`` if parsing the unit results in a ValueError during reading.
+        Default is ``'bunit'``.
+
+    copy : bool, optional
+        Copy the data while creating the `~astropy.nddata.NDData` instance?
+        Default is ``False``.
+
+    dtype : `numpy.dtype`-like or None, optional
+        If not ``None`` the data array is converted to this dtype before
+        returning. See `numpy.ndarray.astype` for more details.
+        Default is ``None``.
+
+    kwargs_for_open :
+        Additional keyword arguments that are passed to
+        :func:`~astropy.io.fits.open` (not all of them might be possible).
+    """
+
+    # Hardcoded values to get additional information about mask and uncertainty
+    kw_hdr_masktype = 'boolean mask'
+    kw_hdr_uncerttype = {
+        'standard deviation uncertainty': StdDevUncertainty,
+        'unknown uncertainty type': UnknownUncertainty}
+
+    with fits_open(filename, mode='readonly', **kwargs_for_open) as hdus:
+        # Read the data and meta from the specified extensions
+        data = hdus[ext_data].data
+        if dtype is not None:
+            data = data.astype(dtype)
+        meta = hdus[ext_meta].header
+
+        # Read the mask and uncertainty from the specified extensions but
+        # silently fail if the extension does not exist.
+        mask = None
+        if ext_mask in hdus:
+            mask = hdus[ext_mask].data
+            # Convert it to boolean array?
+            if kw_hdr_masktype in hdus[ext_mask].header.get('comment', []):
+                mask = mask.astype(bool)
+
+        uncertainty = None
+        if ext_uncert in hdus:
+            uncertainty = hdus[ext_uncert].data
+
+            hdr = hdus[ext_uncert].header
+            # Get the required class for the uncertainty
+            cls = (kw_hdr_uncerttype[kw] for kw in kw_hdr_uncerttype
+                   if kw in hdr.get('comment', []))
+            cls = next(cls, UnknownUncertainty)
+
+            # Get the unit for the uncertainty if present
+            unit_ = hdr[kw_unit].lower() if kw_unit in hdr else None
+
+            # Don't copy it here, if a copy is required do it when creating
+            # NDData.
+            uncertainty = cls(uncertainty, unit=unit_, copy=False)
+
+        # Load unit and wcs from header
+        unit = None
+        if kw_unit is not None and kw_unit in meta:
+            try:
+                unit = u.Unit(meta[kw_unit])
+            except ValueError:
+                # ValueError is raised if the unit isn't convertible to an
+                # astropy unit. Maybe they tried all-uppercase: maybe lowercase
+                # will work
+                unit = u.Unit(meta[kw_unit].lower())
```

Comment:
yes the last one will raise the error. I actually wanted to remove this second conversion using lower-case since with prefixes this might be fragile: `bunit = 'MPC'` would become milli-parsec. :smile:

I wanted to investigate how one could convert the Exception-message and issue it as `log.info` (because units have very informative exceptions if it's not useable) and then forgot about it...